src/inference/src/ie_core.cpp

```diff
@@ -245,32 +245,74 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
-            auto it = config.find(CONFIG_KEY(CACHE_DIR));
-            if (it != config.end()) {
-                std::lock_guard<std::mutex> lock(_cacheConfigMutex);
-                fillConfig(_cacheConfig, it->second.as<std::string>());
-                for (auto& deviceCfg : _cacheConfigPerDevice) {
-                    fillConfig(deviceCfg.second, it->second.as<std::string>());
+                if (it->first == ov::cache_dir.name()) {
+                    std::lock_guard<std::mutex> lock(_cacheConfigMutex);
+                    fillConfig(_cacheConfig, it->second.as<std::string>());
+                    for (auto& deviceCfg : _cacheConfigPerDevice) {
+                        fillConfig(deviceCfg.second, it->second.as<std::string>());
+                    }
+                } else if (it->first == ov::force_tbb_terminate.name()) {
+                    auto flag = it->second.as<std::string>() == CONFIG_VALUE(YES) ? true : false;
+                    executorManager()->setTbbFlag(flag);
+                }
+
+                {
+                    std::lock_guard<std::mutex> lock(_commonConfigMutex);
+                    auto temp = _commonConfig.find(it->first);
+                    if (temp != _commonConfig.end()) {
+                        _commonConfig[it->first] = it->second;
+                    } else {
+                        _commonConfig.insert(*it);
+                    }
+                }
+                it = config.erase(it);
+            }
+        }
+
+        void update_config(const std::string& device_name, ov::AnyMap& config) const {
+            if (device_name.find("BATCH") != std::string::npos) {
+                if (config.find(ov::auto_batch_timeout.name()) == config.end()) {
+                    auto timeout = get_core_config(ov::auto_batch_timeout.name(), false);
+                    if (!timeout.empty())
+                        config.insert({ov::auto_batch_timeout.name(), timeout});
                 }
-                config.erase(it);
             }
+        }
 
-            it = config.find(ov::force_tbb_terminate.name());
-            if (it != config.end()) {
-                auto flag = it->second.as<std::string>() == CONFIG_VALUE(YES) ? true : false;
-                executorManager()->setTbbFlag(flag);
-                config.erase(it);
+        ov::Any get_core_config(const std::string& name, bool use_default_value = true) const {
+            {
+                std::lock_guard<std::mutex> lock(_commonConfigMutex);
+                auto it = _commonConfig.find(name);
+                if (it != _commonConfig.end()) {
+                    return it->second;
+                }
             }
 
-            it = config.find(ov::hint::allow_auto_batching.name());
-            if (it != config.end()) {
-                auto flag = it->second.as<bool>();
-                flag_allow_auto_batching = flag;
-                config.erase(it);
+            if (!use_default_value)
+                return ov::Any();
```

Comment:
when do we need this for batch timeout?

Reply:
If core doesn't set batch_timeout, we don't need set it explicitly with a default value